src/gasless-provider.ts

```diff
@@ -57,32 +53,44 @@ export class GaslessProvider extends ProviderWrapper {
     paymasterClient: Paymaster,
     publicClient: ReturnType<typeof createPublicClient>,
     simpleAccountFactoryAddress: `0x${string}`,
+    smartAccount?: `0x${string}`,
   ) {
     // NOTE: Bundlers can support many entry points, but currently they only support one, we use this method so if they ever add a new one the entry point will still work
     const entryPoint = (await bundlerClient.supportedEntryPoints())[0];
     const owner = privateKeyToAccount(_signerPk);
 
-    const initCode = concat([
-      simpleAccountFactoryAddress,
-      encodeFunctionData({
-        abi: [
-          {
-            inputs: [
-              { name: 'owner', type: 'address' },
-              { name: 'salt', type: 'uint256' },
-            ],
-            name: 'createAccount',
-            outputs: [{ name: 'ret', type: 'address' }],
-            stateMutability: 'nonpayable',
-            type: 'function',
-          },
-        ],
-        args: [owner.address, getRandomBigInt(0n, 2n ** 256n)],
-      }),
-    ]);
-
-    const senderAddress = await getSenderAddress(publicClient, {
-      initCode: initCode,
+    const initCode =
+      smartAccount === undefined
+        ? concat([
+            simpleAccountFactoryAddress,
+            encodeFunctionData({
```

Comment:
You can always use `[...]` instead of concat
```suggestion
        ? [
            simpleAccountFactoryAddress,
            ...encodeFunctionData({...})
            ]
```

Reply:
We need the concat because the type is supposed to be a string not an array